framework/doc/content/syntax/Functors/index.md

```diff
@@ -0,0 +1,138 @@
+a face of an element, and so on. The time arguments represent the state of the functor : current, previous value (whether in time or iteration),
+or value before that previous one.
+
+!alert note
+If a Functor is reported as missing by the simulation, you may use the [!param](/Debug/SetupDebugAction/show_functors)
+to get more information about which functors were created and requested.
+
+## Developing with functors
+
+Functors are evaluated on-the-fly. E.g. they can be viewed as functions of the current location in
+space (and time). Functors provide several overloads of the
+`operator()` method for different "geometric quantities". One example of a
+"geometric quantity" is based around an element, e.g. for an `FVElementalKernel`, the
+value of a functor material property in a cell-averaged sense can be obtained by
+the syntax
+
+- `_foo(makeElemArg(_current_elem), determineState())`
+
+where here `_foo` is a functor data member of the kernel, `makeElemArg` is a helper routine for creating a
+functor element-based spatial argument, and `determineState()` is a helper routine for determining the correct
+time state to evaluate at, e.g. the current time for an implicit kernel and the old time for an explicit kernel.
+
+### Spatial arguments to functors id=spatial-overloads
+
+In the following subsections, we describe the various spatial arguments that functors can be evaluated at.
+Almost no functor developers should have to concern
+themselves with these details as most functor definitions should just appear as functions of
+space and time, e.g. the same lambda defining the property evaluation should apply across all
+spatial and temporal arguments. However, in the case that a functor developer wishes to
+create specific implementations for specific arguments (as illustrated in `IMakeMyOwnFunctorProps`
+test class) or simply wishes to know more about the system, we give the details below.
+
+Any call to a functor looks like the following
+`_foo(const SpatialArg & r, const TemporalArg & t)`. Below are the possible type overloads of
+`SpatialArg`.
+
+#### FaceArg
+
+A typedef defining a "face" evaluation calling argument. This is composed of
+
+- a face information object which defines our location in space
+- a limiter which defines how the functor evaluated on either side of the face should be
+  interpolated to the face
+- a boolean which states whether the face information element is upwind of the face
+- a boolean to indicate whether to correct for element skewness
+- a pair of subdomain IDs. These do not always correspond to the face info element subdomain
+  ID and face info neighbor subdomain ID. For instance if a flux kernel is operating at a
+  subdomain boundary on which the kernel is defined on one side but not the other, the
+  passed-in subdomain IDs will both correspond to the subdomain ID that the flux kernel is
+  defined on
```

Comment:
This has been replaced with a `const Elem * face_side` data member

Reply:
changed